### PR TITLE
Fixed the update application call

### PR DESCRIPTION
**Pull request by nicktate**
* Cleaned up error handling and callback logic
* Removed use of successHandler and split error and regular handlers
* Removed a lot of the wrapper functions because you ended up with the same amount of
boilerplate but it caused you to have to jump to different files and base classes
to follow flow
* Fixed updateApplication to correctly pull existingApp from getApplications
* Cleaned up getApplication contianers to include unloaded
* million other bug fixes

@normanjoyner @jeremykross

### CODE REVIEW

File: lib/kubernetes-api.js [hunk 1]
```diff
@@ -157,106 +170,144 @@ class KubernetesApi extends ApiImplementation {
...
 
-        this.getApplications((err, apps) => {
+                const pods = _.map(util.safeParse(body).items, pod => {
+                    const nodeName = _.get(pod, 'spec.nodeName');
+                    const k8sContainer = _.get(pod, 'spec');
+                    const status = _.get(pod, 'status.phase');
+
+                    // we need a more permanent ID for a container
+                    const containerID = _.get(pod, 'metadata.uid');
+                    //const containerID = _.replace(_.get(pod, 'status.containerStatuses[0].containerID'), 'docker://', '');
```
Comment (normanjoyner) [inline lib/kubernetes-api.js]: blow this away

File: lib/kubernetes-api.js [hunk 2]
```diff
@@ -422,154 +472,121 @@ class KubernetesApi extends ApiImplementation {
...
-                            return handler();
+                            return delete_cb();
                         });
+                    });
+                }
 
-                    } else {
-                        console.log("Trying again in 4 sec");
-                        setTimeout(safeDeleteRC, 4000);
+                return async.retry({ time: 5, interval: 4000 }, safeDeleteRC, (err) => {
```
Comment (normanjoyner) [inline lib/kubernetes-api.js]: should this be `times`?

File: lib/kubernetes-api.js [hunk 3]
```diff
@@ -578,29 +595,48 @@ class KubernetesApi extends ApiImplementation {
...
+            const csLeader = {
+                'id': process.argv[6] || os.hostname(),
+                'host_name': os.hostname(),
 
-                    address: {
-                        private: this.k8sApiIP,
-                        public: this.k8sApiIP
+                'metadata': {
+                    'containership': {
+                        'version': '0.0.0'
```
Comment (normanjoyner) [inline lib/kubernetes-api.js]: any reason to have this?

Comment (nicktate) [inline lib/kubernetes-api.js]: not sure, was already in here
... let me see

**Comment by nicktate** [comment]
@normanjoyner updated

**Pull request merged**